spec/actors/hyku_addons/actors/json_fields_actor_spec.rb

```diff
@@ -0,0 +1,285 @@
+      end
+
+      it "turns fields into null or adds empty arrays" do
+        expect(env.attributes).to eq(transformed_attributes)
+      end
+    end
+
+    context "with no name type" do
+      let(:attributes) do
+        {
+          editor: [
+            { "editor_name_type" => "", "editor_given_name" => "Lihong V.", "editor_family_name" => "Wang" }
+          ]
+        }
+      end
+
+      let(:transformed_attributes) do
+        { "editor" => ["[{\"editor_name_type\":\"\",\"editor_given_name\":\"Lihong V.\",\"editor_family_name\":\"Wang\"}]"], "creator" => [], "contributor" => [], "funder" => [], "alternate_identifier" => [], "related_identifier" => [], "current_he_institution" => [] }
+      end
+
+      it "adds the attributes" do
+        expect(env.attributes).to eq(transformed_attributes)
+      end
+    end
+
+    context "with nested fields" do
+      let(:attributes) do
+        {
+          creator: [
+            { "creator_name_type" => "Personal", "creator_given_name" => "Liang", "creator_family_name" => "Gao" },
+            { "creator_name_type" => "Personal", "creator_given_name" => "Lidai", "creator_family_name" => "Wang" },
+            { "creator_name_type" => "Personal", "creator_given_name" => "Chiye", "creator_family_name" => "Li" },
+            { "creator_name_type" => "Personal", "creator_given_name" => "Yan", "creator_family_name" => "Liu" },
+            creator: [
+              { "creator_name_type" => "Personal", "creator_given_name" => "Haixin", "creator_family_name" => "Ke" },
+              { "creator_name_type" => "Personal", "creator_given_name" => "Chi", "creator_family_name" => "Zhang" },
+              { "creator_name_type" => "Personal", "creator_given_name" => "Lihong V.", "creator_family_name" => "Wang" }
+            ]
+          ]
+        }
+      end
+
+      let(:transformed_attributes) do
+        { "creator" => ["[{\"creator_name_type\":\"Personal\",\"creator_given_name\":\"Liang\",\"creator_family_name\":\"Gao\"},{\"creator_name_type\":\"Personal\",\"creator_given_name\":\"Lidai\",\"creator_family_name\":\"Wang\"},{\"creator_name_type\":\"Personal\",\"creator_given_name\":\"Chiye\",\"creator_family_name\":\"Li\"},{\"creator_name_type\":\"Personal\",\"creator_given_name\":\"Yan\",\"creator_family_name\":\"Liu\"},{\"creator\":[{\"creator_name_type\":\"Personal\",\"creator_given_name\":\"Haixin\",\"creator_family_name\":\"Ke\"},{\"creator_name_type\":\"Personal\",\"creator_given_name\":\"Chi\",\"creator_family_name\":\"Zhang\"},{\"creator_name_type\":\"Personal\",\"creator_given_name\":\"Lihong V.\",\"creator_family_name\":\"Wang\"}]}]"], "contributor" => [], "funder" => [], "alternate_identifier" => [], "related_identifier" => [], "editor" => [], "current_he_institution" => [] }
+      end
+
+      it "recurses through them" do
+        expect(env.attributes).to eq(transformed_attributes)
+      end
+    end
+  end
+
+  describe "#update" do
```

Comment:
I am not sure this is testsing that the work is updated based on the values passed to **env**.  Actor update is supposed to update the actual **work** passed to it with the **attributes** passed to it, so should the expectation not test that the work creator was updated to match the attributes passed to env.

Reply:
OK.  So it sounds like I should create the work itself, outside of the env, then call update on it and test the attributes have been updated correctly.

Reply:
I think what you already have the right setup, just to check in the expectation that the work json fields changed in line new the json fields values passed to the actor